shell/browser/win/dark_mode.cc

```diff
@@ -1,178 +1,51 @@
-// Copyright (c) 2020 Microsoft Inc. All rights reserved.
+// Copyright (c) 2022 Microsoft Inc. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE-CHROMIUM file.
 
 #include "shell/browser/win/dark_mode.h"
 
 #include <dwmapi.h>  // DwmSetWindowAttribute()
 
-#include "base/files/file_path.h"
-#include "base/scoped_native_library.h"
-#include "base/win/pe_image.h"
-#include "base/win/win_util.h"
-#include "base/win/windows_version.h"
+#define DARK_MODE_STRING_NAME L"DarkMode_Explorer"
+#define DWMWA_USE_IMMERSIVE_DARK_MODE_BEFORE_20H1 19
+#define DWMWA_USE_IMMERSIVE_DARK_MODE 20
 
 // This namespace contains code originally from
-// https://github.com/ysc3839/win32-darkmode/
-// governed by the MIT license and (c) Richard Yu
+// https://github.com/microsoft/terminal
+// governed by the MIT license and (c) Microsoft Corporation.
 namespace {
 
-// 1903 18362
-enum PreferredAppMode { Default, AllowDark, ForceDark, ForceLight, Max };
+// Use private Windows API to set window theme.
```

Comment:
What are we considering "private Windows API" at this point? It looks like `DWMWA_USE_IMMERSIVE_DARK_MODE` is documented (although only listed as being valid for Windows 11 builds). Seems the only thing not documented is `L"DarkMode_Explorer"` and `DWMWA_USE_IMMERSIVE_DARK_MODE_BEFORE_20H1`? Let's document it, in the hopes of this not needing to be behind a fuse?

Reply:
Interesting, back when I created the patch there was no Win11, so apparently they [made the value stable](https://docs.microsoft.com/en-us/windows/win32/api/dwmapi/ne-dwmapi-dwmwindowattribute). That makes the API public for Win11, on Win10 I assume we can expect no API breakages at this point - maybe this is already enough reason to drop the fuse (I want to check back with Releases WG to be double-sure). Not sure about `L"DarkMode_Explorer"` .. `Explorer` is listed as [example in the docs](https://docs.microsoft.com/en-us/windows/win32/api/uxtheme/nf-uxtheme-setwindowtheme) - so it's kinda document.